services/housekeeper/housekeeper.go

```diff
@@ -171,6 +179,49 @@ func (hk *Housekeeper) updateProposerDuties(headSlot uint64) {
 	hk.UpdateProposerDutiesWithoutChecks(headSlot)
 }
 
+func (hk *Housekeeper) monitorMevCommitBuilderRegistrations() {
+	newBuilderRegistered, err := hk.mevCommitClient.ListenForActiveBuildersEvents()
+	if err != nil {
+		hk.log.WithError(err).Error("failed to subscribe to MEV-Commit builder registered events")
+		return
+	}
+
+	for {
+		builder := <-newBuilderRegistered
```

Comment:
This is not really good design as if the application is closed then this goroutine is blocked. Ideally there should be some mechanism to close this, like passing a context to the start/having a channel which will be closed on stop and adding a check for this channel here.

Or you can close the newBuilderRegistered channel on close and then check it here:
```
builder, more := <-newBuilderRegistered
if !more {
    return
}
```

or
```
for builder := range newBuilderRegistered {}
```

Reply:
Thanks for the detailed advice. There's no context available to pass at the top-level, I don't think the housekeeper service itself handles shutdowns gracefully.

Reply:
I've opted for the second option you described. I could close the channel, but there's really no single on the application shutdown.